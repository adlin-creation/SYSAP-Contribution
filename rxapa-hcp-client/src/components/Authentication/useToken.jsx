import { useState } from "react";

/**
 *
 * @returns Defines custom hook to manage user token.
 */
export default function useToken() {
  const getToken = () => {
    const tokenString = sessionStorage.getItem("token");
    const userToken = JSON.parse(tokenString);
    return userToken?.token;
  };

  const [token, setToken] = useState(getToken());

  const saveToken = (userToken) => {
    if (userToken) {
      sessionStorage.setItem("token", JSON.stringify(userToken));
      setToken(userToken.token);
    } else {
      sessionStorage.removeItem("token");
      setToken(null);
    }
  };

  return {
    setToken: saveToken,
    token,
  };
}