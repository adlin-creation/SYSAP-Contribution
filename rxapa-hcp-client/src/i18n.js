<<<<<<< HEAD
import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import HttpBackend from "i18next-http-backend";

i18n
  .use(HttpBackend)
  .use(initReactI18next)
  .init({
    lng: localStorage.getItem("i18nextLng") || "fr", //maintenir la langue après refresh, si getItem retourne null alors "fr"
    fallbackLng: "fr", //default language (démarrage en fr)
    ns: [
      "App",
      "Blocs",
      "Cycles",
      "Exercises",
      "Evaluations",
      "Patients",
      "Phases",
      "Professionals",
      "Programs",
      "Sessions",
      "Authentication",
      "Backend",
    ], // un tableau de namespace (ou chercher)
    defaultNS: "App", // Il faut envisager un defaultNS App ne contient que les clé pour le fichier App
    backend: {
      loadPath: "/locales/{{lng}}/{{ns}}.json", // Où chercher les fichiers JSON
    },

    interpolation: { escapeValue: false }, // Pour échaper les caractères spéciaux (limite XSS attacks) exemple : les balises
  }); // sont interprétés pas affichées

export default i18n;
=======
import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import HttpBackend from "i18next-http-backend";

i18n
  .use(HttpBackend)
  .use(initReactI18next)
  .init({
    lng: localStorage.getItem("i18nextLng") || "fr", //maintenir la langue après refresh, si getItem retourne null alors "fr"
    fallbackLng: "fr", //default language (démarrage en fr)
    ns: [
      "App",
      "Blocs",
      "Cycles",
      "Exercises",
      "Evaluations",
      "Patients",
      "Phases",
      "Professionals",
      "Programs",
      "Sessions",
      "Authentication",
      "Backend",
    ], // un tableau de namespace (ou chercher)
    defaultNS: "App", // Il faut envisager un defaultNS App ne contient que les clé pour le fichier App
    backend: {
      loadPath: "/locales/{{lng}}/{{ns}}.json", // Où chercher les fichiers JSON
    },

    interpolation: { escapeValue: false }, // Pour échaper les caractères spéciaux (limite XSS attacks) exemple : les balises
  }); // sont interprétés pas affichées

export default i18n;
>>>>>>> 14cb596907bdb9208979898040b04da5a64dcc68
