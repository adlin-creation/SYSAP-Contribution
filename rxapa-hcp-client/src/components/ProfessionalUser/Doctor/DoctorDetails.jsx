<<<<<<< HEAD
import React, { useEffect } from "react";
import { Row, Col, Input, Button, Form, Switch, Modal as AntModal } from "antd";
import { SaveOutlined } from "@ant-design/icons";
import { Controller, useForm } from "react-hook-form";
import axios from "axios";
import Constants from "../../Utils/Constants";
import useToken from "../../Authentication/useToken";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";

function DoctorDetails({ doctor, onClose, refetchDoctors, openModal }) {
  const { t } = useTranslation("Professionals");
  const {
    handleSubmit,
    control,
    setValue,
    formState: { errors },
  } = useForm();
  const { token } = useToken();

  // Pré-remplir le formulaire avec les données du médecin
  useEffect(() => {
    if (doctor) {
      setValue("firstname", doctor.firstname);
      setValue("lastname", doctor.lastname);
      setValue("email", doctor.email);
      setValue("phoneNumber", doctor.phoneNumber);
      setValue("active", doctor.active);
    }
  }, [doctor, setValue]);

  const onSubmit = (data) => {
    const doctorData = {
      ...data,
      role: "doctor",
    };

    axios
      .put(
        `${Constants.SERVER_URL}/update-professional-user/${doctor.id}`,
        doctorData,
        {
          headers: { Authorization: "Bearer " + token },
        }
      )
      .then(() => {
        refetchDoctors();
        AntModal.success({
          content: t("Physicians.success_updating_msg"),
          okText: t("Physicians.button_close"),
          centered: true,
          onOk: () => {
            onClose();
          },
        });
      })
      .catch((err) => {
        const errorMessage =
          t(`Backend:${err.response?.data?.message}`) ||
          t("Physicians.error_updating_physician");
        AntModal.error({
          content: errorMessage,
          okText: t("Physicians.button_close"),
          centered: true,
        });
      });
  };

  return (
    <Row justify="center">
      <Col span={16}>
        <Form layout="vertical" onFinish={handleSubmit(onSubmit)}>
          <Row gutter={16}>
            {/* First Name and Last Name fields */}
            <Col span={12}>
              <Form.Item
                label={t("Physicians.label_first_name")}
                required
                validateStatus={errors.firstname ? "error" : ""}
                help={errors.firstname?.message}
              >
                <Controller
                  name="firstname"
                  control={control}
                  rules={{
                    required: t("Physicians.error_required_first_name"),
                    minLength: {
                      value: 2,
                      message: t("Physicians.error_first_name_min_length"),
                    },
                  }}
                  render={({ field }) => <Input {...field} />}
                />
              </Form.Item>
            </Col>
            <Col span={12}>
              <Form.Item
                label={t("Physicians.label_last_name")}
                required
                validateStatus={errors.lastname ? "error" : ""}
                help={errors.lastname?.message}
              >
                <Controller
                  name="lastname"
                  control={control}
                  rules={{
                    required: t("Physicians.error_required_last_name"),
                    minLength: {
                      value: 2,
                      message: t("Physicians.error_last_name_min_length"),
                    },
                  }}
                  render={({ field }) => <Input {...field} />}
                />
              </Form.Item>
            </Col>
          </Row>

          <Row gutter={16}>
            {/* Email and Phone Number fields */}
            <Col span={12}>
              <Form.Item
                label={t("Physicians.label_email")}
                required
                validateStatus={errors.email ? "error" : ""}
                help={errors.email?.message}
              >
                <Controller
                  name="email"
                  control={control}
                  rules={{
                    required: t("Physicians.error_required_email"),
                    pattern: {
                      value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                      message: t("Physicians.error_invalid_email_format"),
                    },
                  }}
                  render={({ field }) => <Input type="email" {...field} />}
                />
              </Form.Item>
            </Col>
            <Col span={12}>
              <Form.Item
                label={t("Physicians.label_phone_number")}
                required
                validateStatus={errors.phoneNumber ? "error" : ""}
                help={errors.phoneNumber?.message}
              >
                <Controller
                  name="phoneNumber"
                  control={control}
                  rules={{
                    required: t("Physicians.error_required_phone_number"),
                    pattern: {
                      value: /^[0-9+\s-]{8,}$/,
                      message: t("Physicians.error_invalid_phone_number"),
                    },
                  }}
                  render={({ field }) => <Input {...field} />}
                />
              </Form.Item>
            </Col>
          </Row>

          <Row gutter={16}>
            {/* Status field */}
            <Col span={12}>
              <Form.Item label={t("Physicians.label_status")}>
                <Controller
                  name="active"
                  control={control}
                  render={({ field: { value, onChange } }) => (
                    <Switch
                      checked={value}
                      onChange={onChange}
                      checkedChildren={t("Physicians.active_status")}
                      unCheckedChildren={t("Physicians.inactive_status")}
                    />
                  )}
                />
              </Form.Item>
            </Col>
          </Row>

          <Form.Item>
            <Button type="primary" htmlType="submit" icon={<SaveOutlined />}>
              {t("Physicians.button_update_physician")}
            </Button>
          </Form.Item>
        </Form>
      </Col>
    </Row>
  );
}

DoctorDetails.propTypes = {
  doctor: PropTypes.object.isRequired,
  onClose: PropTypes.func.isRequired,
  refetchDoctors: PropTypes.func.isRequired,
  openModal: PropTypes.func.isRequired,
};

export default DoctorDetails;
=======
import React, { useEffect } from "react";
import { Row, Col, Input, Button, Form, Switch, Modal as AntModal } from "antd";
import { SaveOutlined } from "@ant-design/icons";
import { Controller, useForm } from "react-hook-form";
import axios from "axios";
import Constants from "../../Utils/Constants";
import useToken from "../../Authentication/useToken";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";

function DoctorDetails({ doctor, onClose, refetchDoctors, openModal }) {
  const { t } = useTranslation("Professionals");
  const {
    handleSubmit,
    control,
    setValue,
    formState: { errors },
  } = useForm();
  const { token } = useToken();

  // Pré-remplir le formulaire avec les données du médecin
  useEffect(() => {
    if (doctor) {
      setValue("firstname", doctor.firstname);
      setValue("lastname", doctor.lastname);
      setValue("email", doctor.email);
      setValue("phoneNumber", doctor.phoneNumber);
      setValue("active", doctor.active);
    }
  }, [doctor, setValue]);

  const onSubmit = (data) => {
    const doctorData = {
      ...data,
      role: "doctor",
    };

    axios
      .put(
        `${Constants.SERVER_URL}/update-professional-user/${doctor.id}`,
        doctorData,
        {
          headers: { Authorization: "Bearer " + token },
        }
      )
      .then(() => {
        refetchDoctors();
        AntModal.success({
          content: t("Physicians.success_updating_msg"),
          okText: t("Physicians.button_close"),
          centered: true,
          onOk: () => {
            onClose();
          },
        });
      })
      .catch((err) => {
        const errorMessage =
          t(`Backend:${err.response?.data?.message}`) ||
          t("Physicians.error_updating_physician");
        AntModal.error({
          content: errorMessage,
          okText: t("Physicians.button_close"),
          centered: true,
        });
      });
  };

  return (
    <Row justify="center">
      <Col span={16}>
        <Form layout="vertical" onFinish={handleSubmit(onSubmit)}>
          <Row gutter={16}>
            {/* First Name and Last Name fields */}
            <Col span={12}>
              <Form.Item
                label={t("Physicians.label_first_name")}
                required
                validateStatus={errors.firstname ? "error" : ""}
                help={errors.firstname?.message}
              >
                <Controller
                  name="firstname"
                  control={control}
                  rules={{
                    required: t("Physicians.error_required_first_name"),
                    minLength: {
                      value: 2,
                      message: t("Physicians.error_first_name_min_length"),
                    },
                  }}
                  render={({ field }) => <Input {...field} />}
                />
              </Form.Item>
            </Col>
            <Col span={12}>
              <Form.Item
                label={t("Physicians.label_last_name")}
                required
                validateStatus={errors.lastname ? "error" : ""}
                help={errors.lastname?.message}
              >
                <Controller
                  name="lastname"
                  control={control}
                  rules={{
                    required: t("Physicians.error_required_last_name"),
                    minLength: {
                      value: 2,
                      message: t("Physicians.error_last_name_min_length"),
                    },
                  }}
                  render={({ field }) => <Input {...field} />}
                />
              </Form.Item>
            </Col>
          </Row>

          <Row gutter={16}>
            {/* Email and Phone Number fields */}
            <Col span={12}>
              <Form.Item
                label={t("Physicians.label_email")}
                required
                validateStatus={errors.email ? "error" : ""}
                help={errors.email?.message}
              >
                <Controller
                  name="email"
                  control={control}
                  rules={{
                    required: t("Physicians.error_required_email"),
                    pattern: {
                      value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                      message: t("Physicians.error_invalid_email_format"),
                    },
                  }}
                  render={({ field }) => <Input type="email" {...field} />}
                />
              </Form.Item>
            </Col>
            <Col span={12}>
              <Form.Item
                label={t("Physicians.label_phone_number")}
                required
                validateStatus={errors.phoneNumber ? "error" : ""}
                help={errors.phoneNumber?.message}
              >
                <Controller
                  name="phoneNumber"
                  control={control}
                  rules={{
                    required: t("Physicians.error_required_phone_number"),
                    pattern: {
                      value: /^[0-9+\s-]{8,}$/,
                      message: t("Physicians.error_invalid_phone_number"),
                    },
                  }}
                  render={({ field }) => <Input {...field} />}
                />
              </Form.Item>
            </Col>
          </Row>

          <Row gutter={16}>
            {/* Status field */}
            <Col span={12}>
              <Form.Item label={t("Physicians.label_status")}>
                <Controller
                  name="active"
                  control={control}
                  render={({ field: { value, onChange } }) => (
                    <Switch
                      checked={value}
                      onChange={onChange}
                      checkedChildren={t("Physicians.active_status")}
                      unCheckedChildren={t("Physicians.inactive_status")}
                    />
                  )}
                />
              </Form.Item>
            </Col>
          </Row>

          <Form.Item>
            <Button type="primary" htmlType="submit" icon={<SaveOutlined />}>
              {t("Physicians.button_update_physician")}
            </Button>
          </Form.Item>
        </Form>
      </Col>
    </Row>
  );
}

DoctorDetails.propTypes = {
  doctor: PropTypes.object.isRequired,
  onClose: PropTypes.func.isRequired,
  refetchDoctors: PropTypes.func.isRequired,
  openModal: PropTypes.func.isRequired,
};

export default DoctorDetails;
>>>>>>> 14cb596907bdb9208979898040b04da5a64dcc68
