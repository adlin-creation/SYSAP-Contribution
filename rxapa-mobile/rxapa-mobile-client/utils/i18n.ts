import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import * as RNLocalize from 'react-native-localize';

import en_home from '../locales/en/Home.json';
import en_index from '../locales/en/Index.json';
import en_layout from '../locales/en/Layout.json';
import en_cahier from '../locales/en/CahierDeSuivi.json';
import en_config from '../locales/en/Configuration.json';
import en_programme from '../locales/en/Programme.json';
import en_progression from '../locales/en/Progression.json';
import en_seance from '../locales/en/Seance.json';

import fr_home from '../locales/fr/Home.json'
import fr_index from '../locales/fr/Index.json'
import fr_layout from '../locales/fr/Layout.json';
import fr_cahier from '../locales/fr/CahierDeSuivi.json';
import fr_config from '../locales/fr/Configuration.json';
import fr_programme from '../locales/fr/Programme.json';
import fr_progression from '../locales/fr/Progression.json';
import fr_seance from '../locales/fr/Seance.json';

let locales = [];
try {
    locales = RNLocalize.getLocales();
} catch (error) {
    console.error('Error fetching locales:', error);
    locales = [{ languageCode: 'fr' }];
}
const defaultLang = locales[0]?.languageCode || 'fr';

i18n
  .use(initReactI18next)
  .init({
    resources: {
        en: { 
            Home: en_home,
            Index: en_index,
            Layout: en_layout,
            Cahier: en_cahier,
            Config: en_config,
            Programme: en_programme,
            Progression: en_progression,
            Seance: en_seance
        },

        fr: {
            Home: fr_home,
            Index: fr_index,
            Layout: fr_layout,
            Cahier: fr_cahier,
            Config: fr_config,
            Programme: fr_programme,
            Progression: fr_progression,
            Seance: fr_seance
        },
    },
    lng: defaultLang,
    fallbackLng: "fr",
    ns: [
        "Home",
        "Index",
        "Layout",
        "Cahier",
        "Programme",
        "Progression",
        "Seance"
    ],
    defaultNS: "Home",
    interpolation: { escapeValue: false },
  });

export default i18n;
