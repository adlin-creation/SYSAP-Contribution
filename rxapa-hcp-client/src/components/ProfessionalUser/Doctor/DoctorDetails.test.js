// DoctorDetails.test.js

import React, { Suspense } from "react";
import renderer from "react-test-renderer";
import DoctorDetails from "./DoctorDetails";
import useToken from "../../Authentication/useToken";
import { I18nextProvider } from 'react-i18next';
import i18n from '../../../i18n';

jest.mock("../../Authentication/useToken", () => ({
  __esModule: true,
  default: jest.fn(),
}));

beforeAll(() => {
  // Mock matchMedia (Ant Design might use for responsive layout)
  global.matchMedia = jest.fn().mockImplementation(() => ({
    matches: false,
    addListener: jest.fn(),
    removeListener: jest.fn(),
  }));
});

afterAll(() => {
  global.matchMedia.mockRestore();
});

const ErrorBoundary = ({ children }) => {
  return children;
};

test("renders DoctorDetails form snapshot", () => {
  // Mock token
  useToken.mockReturnValue({ token: "fake-token" });

  // Simulate a doctor object
  const mockDoctor = {
    id: 123,
    firstname: "John",
    lastname: "Doe",
    email: "john@example.com",
    phoneNumber: "12345678",
    active: true,
  };

  const mockOnClose = jest.fn();
  const mockRefetchDoctors = jest.fn();
  const mockOpenModal = jest.fn();

  // Render via react-test-renderer
  const component = renderer.create(
    <I18nextProvider i18n={i18n}>
      <ErrorBoundary>
        <Suspense fallback={<div>Loading...</div>}>
          <DoctorDetails
            doctor={mockDoctor}
            onClose={mockOnClose}
            refetchDoctors={mockRefetchDoctors}
            openModal={mockOpenModal}
          />
        </Suspense>
      </ErrorBoundary>
    </I18nextProvider>
  );

  // Snapshot
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
