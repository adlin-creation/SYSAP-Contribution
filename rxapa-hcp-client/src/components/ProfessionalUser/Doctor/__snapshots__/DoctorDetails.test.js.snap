// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DoctorDetails form snapshot 1`] = `
<div
  className="ant-row ant-row-center css-dev-only-do-not-override-1hpnbz2"
  style={
    Object {
      "rowGap": undefined,
    }
  }
>
  <div
    className="ant-col ant-col-16 css-dev-only-do-not-override-1hpnbz2"
    style={Object {}}
  >
    <form
      className="ant-form ant-form-vertical css-dev-only-do-not-override-1hpnbz2"
      onReset={[Function]}
      onSubmit={[Function]}
      style={Object {}}
    >
      <div
        className="ant-row css-dev-only-do-not-override-1hpnbz2"
        style={
          Object {
            "marginLeft": -8,
            "marginRight": -8,
            "rowGap": undefined,
          }
        }
      >
        <div
          className="ant-col ant-col-12 css-dev-only-do-not-override-1hpnbz2"
          style={
            Object {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <div
            className="ant-form-item css-dev-only-do-not-override-1hpnbz2"
          >
            <div
              className="ant-row ant-form-item-row css-dev-only-do-not-override-1hpnbz2"
              style={
                Object {
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="ant-col ant-form-item-label css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <label
                  className="ant-form-item-required"
                  title="Professionals:Doctors:first_name_label"
                >
                  Professionals:Doctors:first_name_label
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input css-dev-only-do-not-override-1hpnbz2 ant-input-outlined"
                      disabled={false}
                      name="firstname"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onCompositionEnd={[Function]}
                      onCompositionStart={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      style={Object {}}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col ant-col-12 css-dev-only-do-not-override-1hpnbz2"
          style={
            Object {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <div
            className="ant-form-item css-dev-only-do-not-override-1hpnbz2"
          >
            <div
              className="ant-row ant-form-item-row css-dev-only-do-not-override-1hpnbz2"
              style={
                Object {
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="ant-col ant-form-item-label css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <label
                  className="ant-form-item-required"
                  title="Professionals:Doctors:last_name_label"
                >
                  Professionals:Doctors:last_name_label
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input css-dev-only-do-not-override-1hpnbz2 ant-input-outlined"
                      disabled={false}
                      name="lastname"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onCompositionEnd={[Function]}
                      onCompositionStart={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      style={Object {}}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row css-dev-only-do-not-override-1hpnbz2"
        style={
          Object {
            "marginLeft": -8,
            "marginRight": -8,
            "rowGap": undefined,
          }
        }
      >
        <div
          className="ant-col ant-col-12 css-dev-only-do-not-override-1hpnbz2"
          style={
            Object {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <div
            className="ant-form-item css-dev-only-do-not-override-1hpnbz2"
          >
            <div
              className="ant-row ant-form-item-row css-dev-only-do-not-override-1hpnbz2"
              style={
                Object {
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="ant-col ant-form-item-label css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <label
                  className="ant-form-item-required"
                  title="Professionals:Doctors:email"
                >
                  Professionals:Doctors:email
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input css-dev-only-do-not-override-1hpnbz2 ant-input-outlined"
                      disabled={false}
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onCompositionEnd={[Function]}
                      onCompositionStart={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      style={Object {}}
                      type="email"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col ant-col-12 css-dev-only-do-not-override-1hpnbz2"
          style={
            Object {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <div
            className="ant-form-item css-dev-only-do-not-override-1hpnbz2"
          >
            <div
              className="ant-row ant-form-item-row css-dev-only-do-not-override-1hpnbz2"
              style={
                Object {
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="ant-col ant-form-item-label css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <label
                  className="ant-form-item-required"
                  title="Professionals:Doctors:phone_number"
                >
                  Professionals:Doctors:phone_number
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input css-dev-only-do-not-override-1hpnbz2 ant-input-outlined"
                      disabled={false}
                      name="phoneNumber"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onCompositionEnd={[Function]}
                      onCompositionStart={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      style={Object {}}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row css-dev-only-do-not-override-1hpnbz2"
        style={
          Object {
            "marginLeft": -8,
            "marginRight": -8,
            "rowGap": undefined,
          }
        }
      >
        <div
          className="ant-col ant-col-12 css-dev-only-do-not-override-1hpnbz2"
          style={
            Object {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <div
            className="ant-form-item css-dev-only-do-not-override-1hpnbz2"
          >
            <div
              className="ant-row ant-form-item-row css-dev-only-do-not-override-1hpnbz2"
              style={
                Object {
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="ant-col ant-form-item-label css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <label
                  className=""
                  title="Professionals:Doctors:status"
                >
                  Professionals:Doctors:status
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control css-dev-only-do-not-override-1hpnbz2"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <button
                      aria-checked={false}
                      className="ant-switch css-dev-only-do-not-override-1hpnbz2"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="switch"
                      style={Object {}}
                      type="button"
                    >
                      <div
                        className="ant-switch-handle"
                      />
                      <span
                        className="ant-switch-inner"
                      >
                        <span
                          className="ant-switch-inner-checked"
                        >
                          Professionals:Doctors:active_status
                        </span>
                        <span
                          className="ant-switch-inner-unchecked"
                        >
                          Professionals:Doctors:inactive_status
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-form-item css-dev-only-do-not-override-1hpnbz2"
      >
        <div
          className="ant-row ant-form-item-row css-dev-only-do-not-override-1hpnbz2"
          style={
            Object {
              "rowGap": undefined,
            }
          }
        >
          <div
            className="ant-col ant-form-item-control css-dev-only-do-not-override-1hpnbz2"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <button
                  className="ant-btn css-dev-only-do-not-override-1hpnbz2 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
                  disabled={false}
                  onClick={[Function]}
                  style={Object {}}
                  type="submit"
                >
                  <span
                    className="ant-btn-icon"
                    style={Object {}}
                  >
                    <span
                      aria-label="save"
                      className="anticon anticon-save"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="save"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M893.3 293.3L730.7 130.7c-7.5-7.5-16.7-13-26.7-16V112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V338.5c0-17-6.7-33.2-18.7-45.2zM384 184h256v104H384V184zm456 656H184V184h136v136c0 17.7 14.3 32 32 32h320c17.7 0 32-14.3 32-32V205.8l136 136V840zM512 442c-79.5 0-144 64.5-144 144s64.5 144 144 144 144-64.5 144-144-64.5-144-144-144zm0 224c-44.2 0-80-35.8-80-80s35.8-80 80-80 80 35.8 80 80-35.8 80-80 80z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span>
                    Professionals:Doctors:update_doctor_button
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
